backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/controllers/BookToOrderController.java

```diff
@@ -0,0 +1,53 @@
+package com.nokia.library.nokiainnovativeproject.controllers;
+
+import com.nokia.library.nokiainnovativeproject.DTOs.BookToOrderDTO;
+import com.nokia.library.nokiainnovativeproject.services.BookToOrderService;
+import com.nokia.library.nokiainnovativeproject.utils.Mappings;
+import com.nokia.library.nokiainnovativeproject.utils.MessageInfo;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.Arrays;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping(Mappings.API_VERSION + Mappings.BOOK_TO_ORDER)
+public class BookToOrderController {
+
+    private final BookToOrderService bookToOrderService;
+
+    @GetMapping(Mappings.GET_ALL)
+    public ResponseEntity getAllBookToOrder(){
+        return MessageInfo.success(bookToOrderService.getAllBookToOrders(), Arrays.asList("list of books to order"));
+    }
+
+    @GetMapping(Mappings.GET_ONE)
+    public ResponseEntity getBookToOrderById(@PathVariable Long id) {
+        return MessageInfo.success(bookToOrderService.getBookToOrderById(id),
+                Arrays.asList("BookToOrderController to order of ID = " + id.toString()));
+    }
+
+    @PostMapping(Mappings.CREATE)
+    public ResponseEntity createBookToOrder(@RequestBody @Valid BookToOrderDTO bookToOrderDTO, BindingResult bindingResult) {
+        ResponseEntity errors = MessageInfo.getErrors(bindingResult);
+        return errors != null ? errors : MessageInfo.success(bookToOrderService.createBookToOrder(bookToOrderDTO),
+                Arrays.asList("BookToOrder created successfully"));
```

Comment:
Please check it out
https://github.com/nokia-wroclaw/innovativeproject-nokia-library/pull/37#discussion_r240944860